src/Huntress/Plugin/Remind.php

```diff
@@ -169,16 +179,40 @@ public static function getHelp(): string
 
     }
 
-    private static function addReminder(Message $message, Carbon $time, string $text)
+    public static function deleteReminder(EventData $data, string $code): ?Promise
+    {
+        /** @var \Doctrine\DBAL\Connection $db */
+        $message = $data->message;
+        $db = $message->client->db;
+        $snow = Snowflake::parse($code);
+        $reminder = $db->executeQuery("SELECT * FROM remind WHERE (`snow` = ?)", [$snow])->fetch();
+
+        if (!$reminder) {
+            return $message->channel->send("No reminder matching `$code` was found.");
+        }
+        if ($message->member->id !== $reminder['idMember']) {
+            $p = new Permission('p.reminder.delete', $data->huntress, false);
+            if (!$p->resolve()) {
+                return $message->channel->send("You cannot delete a reminder created by another user.");
+            }
```

Comment:
this will need a context applied, probably `$p->addMessageContext($data->message);` but my brain is made of cottage cheese rn. most of the other plugins should have it, uhhh first off the top of my head is Roles

Reply:
✔️